core/src/kmx/kmx_plus.cpp

```diff
@@ -386,11 +387,35 @@ COMP_KMXPLUS_ELEM::getElementList(KMX_DWORD elementNumber, KMX_DWORD &length) co
 
 std::u16string
 COMP_KMXPLUS_ELEM_ELEMENT::get_element_string() const {
-  assert((flags & LDML_ELEM_FLAGS_TYPE) == LDML_ELEM_FLAGS_TYPE_CHAR); // should only be called on char
+  assert(type() == LDML_ELEM_FLAGS_TYPE_CHAR); // should only be called on char
   return COMP_KMXPLUS_STRS::str_from_char(element);
 }
 
-// Note: shared with subclass COMP_KMXPLUS_BKSP
+std::deque<std::u32string>
+COMP_KMXPLUS_ELEM_ELEMENT::loadAsStringList(KMX_DWORD length, const COMP_KMXPLUS_STRS &strs) const {
+  std::deque<std::u32string> list;
+  for (KMX_DWORD i = 0; i<length; i++) {
+    const auto &o = this[i];
+    std::u32string str;
+    if (o.type() == LDML_ELEM_FLAGS_TYPE_STR) {
+      // fetch the string
+      const auto str16 = strs.get(o.element);
+      str = km::kbp::kmx::u16string_to_u32string(str16);
+    } else {
```

Comment:
Realized this should assert that the type is `LDML_ELEM_FLAGS_TYPE_UCHR` in case kmc is sending us something random (for a followon)